### PR TITLE
Fix reporting missing required flags

### PR DESCRIPTION
**Pull request by alexmcmanus**
Fixes https://github.com/sindresorhus/meow/issues/165. Also updated readme.md. 

Note that I've made a transitive dependency explicit (decamelize). Rather than add the latest version, I've stuck with the older version brought in by decamelize-keys to avoid unnecessary bloat.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -50,7 +51,14 @@ const getMissingRequiredFlags = (flags, receivedFlags, input) => {
...
 	for (const flag of missingRequiredFlags) {
-		console.error(`\t--${flag.key}${flag.alias ? `, -${flag.alias}` : ''}`);
+		console.error(`\t--${decamelize(flag.key, '-')}${flag.alias ? `, -${flag.alias}` : ''}`);
+	}
+};
+
+const validateOptions = ({flags}) => {
+	const invalidFlags = Object.keys(flags).filter(flagKey => flagKey.includes('-') && flagKey !== '--');
+	if (invalidFlags.length > 0) {
+		throw new Error('Flag keys may not contain \'-\': ' + invalidFlags.join(', '));
```
Comment (sindresorhus) [inline index.js:61]:
```suggestion
		throw new Error(`Flag keys may not contain \`-\`: ${invalidFlags.join(', ')}`);
```

**Comment by sindresorhus** [comment]
You need to update the doc comments in index.d.ts too.

**Comment by MichaelPeter** [comment]
FYI: This breaking change just broke chromatic since chromatic was auto upgrade

https://github.com/chromaui/chromatic-cli/issues/235

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/meow/issues/173

**Pull request merged**